packages/mui-joy/src/Link/Link.tsx

```diff
@@ -0,0 +1,231 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import {
+  unstable_capitalize as capitalize,
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import Typography from '../Typography';
+import linkClasses, { getLinkUtilityClass } from './linkClasses';
+import { LinkProps, LinkTypeMap } from './LinkProps';
+
+const useUtilityClasses = (ownerState: LinkProps) => {
+  const { component, level, color, variant, underline, focusVisible, disabled } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      level,
+      underline && `underline${capitalize(underline)}`,
+      variant && `variant${capitalize(variant)}`,
+      component === 'button' && 'button',
+    ],
+  };
+
+  return composeClasses(slots, getLinkUtilityClass, {});
+};
+
+const LinkRoot = styled(Typography, {
+  name: 'MuiLink',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: LinkProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.underline === 'none' && {
+        textDecoration: 'none',
+      }),
+      ...(ownerState.underline === 'hover' && {
+        textDecoration: 'none',
+        '&:hover': {
+          textDecoration: 'underline',
+        },
+      }),
+      ...(ownerState.underline === 'always' && {
+        textDecoration: 'underline',
+        '&:hover': {
+          textDecorationColor: 'inherit',
+        },
+      }),
+      // Same reset as ButtonBase.root
+      ...(ownerState.component === 'button' && {
+        position: 'relative',
+        WebkitTapHighlightColor: 'transparent',
+        backgroundColor: 'transparent', // Reset default value
+        // We disable the focus ring for mouse, touch and keyboard users.
+        outline: 0,
+        border: 0,
+        margin: 0, // Remove the margin in Safari
+        borderRadius: 0,
+        padding: 0, // Remove the padding in Firefox
```

Comment:
```suggestion
        margin: 0, // Remove the margin in Safari
        borderRadius: 0,
        padding: 0, // Remove the padding in Firefox
        ...(!!ownerState.variant && {
          paddingInline: '0.25em', // better than left, right because it also works with writing mode.
          marginInline: '-0.25em',
        }),
```

I think this is worth trying because the current variant looks so tight which I think anyone will customize it anyway.

<img width="79" alt="image" src="https://user-images.githubusercontent.com/18292247/155300051-d3d4b16e-5e0c-49a6-9c7c-fceceee290cc.png">

Adding paddingInline gives this result:

<img width="83" alt="image" src="https://user-images.githubusercontent.com/18292247/155300377-3ffcf900-cb86-4150-955f-3a0e5d992b51.png">

marginInline is for making the content align with others by default:

<img width="66" alt="image" src="https://user-images.githubusercontent.com/18292247/155300629-fd04cc53-c7bc-4fbe-ad10-0bf3347cd148.png">

Reply:
From what I tested, `0.25em` works best.

Reply:
Agree. Resolved.